src/main/java/scmspain/karyon/restrouter/RouteHandlerProvider.java

```diff
@@ -0,0 +1,38 @@
+package scmspain.karyon.restrouter;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.hystrix.contrib.rxnetty.metricsstream.HystrixMetricsStreamHandler;
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+
+public class RouteHandlerProvider implements Provider<RequestHandler<ByteBuf, ByteBuf>> {
```

Comment:
imo, we are mixing two responsibilities and thats why the name is confusing...

1- We need a "decision manager" to know based in some conditions what RequestHandler provide (actually the condition is simple but maybe someday will not be)
2- A generic provider to return instances of something that implements RequestHandler.

;)